test/unit/nedap/utils/test/api.cljc

```diff
@@ -110,3 +113,69 @@
       (gensym)         42
       (gensym)         nil
       [[1 (gensym) 2]] [[1 (gensym) 3]])))
+
+(deftest expect
+  (let [a (atom 0)]
+    (sut/expect (swap! a inc)
+                :to-change @a
+                :from 0
+                :to 1)
+
+    (sut/expect (swap! a inc)
+                (swap! a inc)
+                :to-change @a
+                :from 1
+                :to 3))
+
+  (testing "the macroexpansion evaluates each part exactly once"
+    (let [proof (atom [])]
+      (sut/expect
+       (swap! proof conj :body)
+       :to-change (do (swap! proof conj :to-change) true)
+       :from      (do (swap! proof conj :from)      true)
+       :to        (do (swap! proof conj :to)        true))
+
+      (is (= [:to-change :from :body :to-change :to]
+             @proof))))
+
+  (testing "normal exception behaviour"
+    (is (thrown-with-msg? #?(:clj ExceptionInfo :cljs js/Error) #"my special failure"
+                          (sut/expect (throw (ex-info "my special failure" {})) :to-change 0 :from 0 :to 0))))
+
+  (testing "failures"
+    (let [test-result (atom {})
+          a (atom 0)]
+      (are [form expected] (match? expected
+                                   (with-redefs [do-report (partial reset! test-result)]
+                                     form
+                                     @test-result))
+        (sut/expect 0 :to-change 0 :from 0 :to 1)
+        '{:type :fail, :expected (clojure.core/= 0 1), :actual (not (clojure.core/= 0 1))}
+
+        (sut/expect (swap! a inc) :to-change @a :from 1 :to 2)
+        '{:type :fail, :expected (clojure.core/= (clojure.core/deref a) 2), :actual (not (clojure.core/= 1 2))}
+
+        (sut/expect (swap! a inc) :to-change @a :from 1 :to 3)
+        '{:type :fail, :expected (clojure.core/= (clojure.core/deref a) 3), :actual (not (clojure.core/= 2 3))})))
+
+  #?(:clj
```

Comment:
So, finally what's the plan for exercising the red path in cljs?

Reply:
The macroexpansion, and thus these assertions, are always run in clojure. So these clj-only assertions should suffice.

The red-path for cljs/clj impl is shared on L146-L159 :)

Reply:
👍 

Then wrap in `letfn` in `(testing "macroexpansion-time validation")`, making it evident that cljs is covered